src/app.ts

```diff
@@ -165,20 +179,59 @@ export default (
       method: 'updateSecret',
     });
 
+    // I think for now we can use the taskName Prefix and edit it when we
+    // are adding additional information to the check
+    const checkResponse = await context.octokit.checks.create({
+      owner: context.payload.organization.login,
+      repo: '.github',
+      name: 'peribolos-run',
+      head_sha: context.payload.after,
+      status: 'queued',
+    });
+
     // Trigger taskrun to apply config changes to org
-    wrapOperationWithMetrics(
-      useApi(APIS.CustomObjectsApi).createNamespacedCustomObject(
-        'tekton.dev',
-        'v1beta1',
-        getNamespace(),
-        'taskruns',
-        generateTaskPayload('peribolos-run', context)
-      ),
+    createTaskRun('peribolos-run', context, [
       {
-        install: context.payload.installation.id,
-        method: 'schedulePushTask',
-      }
-    );
+        name: 'CHECK_RUN_ID',
+        value: checkResponse.data.id.toString(),
+      },
+    ]);
+  });
+
+  app.on('check_run.rerequested', async (context: any) => {
+    // In the future a check is needed if this is the peribolos-run check
+    const checkCommit = context.payload.check_run.head_sha;
+    const headCommit = (
+      await context.octokit.repos.getBranch({
+        owner: context.payload.organization.login,
+        repo: '.github',
+        branch: (
+          await context.octokit.repos.get({
+            owner: context.payload.organization.login,
+            repo: '.github',
+          })
+        ).data.default_branch,
+      })
+    ).data.commit.sha;
+
+    if (checkCommit === headCommit) {
+      createTaskRun('peribolos-run', context, [
```

Comment:
Please reset the check back to pending here.

Reply:
Resolved in the pushed commit.